### DISCUSSION TITLE
Is it possible to adjust capacitors webview configuration?

### DISCUSSION THREAD

**Discussion by xzilja** [ionic-team/capacitor#4196] [Q&A] [answered]
I was looking for native methods to set some configuration on capacitors webview i.e. https://developer.apple.com/documentation/webkit/wkwebviewconfiguration#//apple_ref/doc/c_ref/WKWebViewConfiguration in ios and equivalent in android.

But I don't see anything atm. I'm not looking for a plugin that exposes this to js, rather something I can do in my AppDelegate before app initis.

Is this exposed somehow in capacitor 2 / 3?

**Comment by jcesarmobile** [comment, marked as answer]
In Capacitor 3 you can subclass `CAPBridgeViewController` and override [`webViewConfiguration(for:)`](https://github.com/ionic-team/capacitor/blob/bd59fa3637ec32280864cee4d649267274109e5a/ios/Capacitor/Capacitor/CAPBridgeViewController.swift#L101) to modify the configuration before the web view gets created.

**Reply by lincolnthree** [reply]
On android, I believe this can be done in your MainActivity.java class:

```
package com.maritlabs.topdecked.mtg;

import android.os.Build;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.webkit.WebSettings;
import android.webkit.WebView;

import androidx.annotation.Nullable;

import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {
  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  }

  @Override
  public void onPostCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
    super.onPostCreate(savedInstanceState, persistentState);

    WebView webView = getBridge().getWebView();
    WebSettings settings = webView.getSettings();

   // Customize things here...
  }
}

```

**Reply by xzilja** [reply]
@jcesarmobile I glanced at `webViewConfiguration(for:)` and it is missing config I was hoping to apply for `tintColor`. I've tried handling this in css realm, but safari has weird bug where it changes to default tint color (blue) if input has transform matrix. I wanted to see if this can be solved by setting wkwebviews tintColor perhaps.

Also wanted to ask what would be right place / how to correctly extend `CAPBridgeViewController` to set this. My initial instinct was AppDelegate, but I believe that is not correct (I have limited knowledge in native realm)

**Reply by jcesarmobile** [reply]
`webViewConfiguration` is for the `WKWebViewConfiguration`, that can only be set on `WKWebView` creation.
Any other `WKWebView` properties can be changed at runtime, that's already possible at the moment from any plugin as they have access to the `WKWebView`.
On Capacitor 3 you'll also be able to provide your own `WKWebView` subclass if needed, but shouldn't be needed for setting the `tintColor`.
